spotinst_sdk2/clients/elastigroup/__init__.py

```diff
@@ -1866,6 +1847,116 @@ def detach_elastigroup_instances(self, group_id, detach_configuration):
+        """
+        Fetch a group's Elastilog
+
+        # Arguments
+        group_id (String): Elastigroup ID
+        start_date (String): Get items on or after this date (ISO 8601 or Unix timestamp)
+        end_date (String): Get items on or before this date (ISO 8601 or Unix timestamp)
+        limit (Integer): Maximum number of items to return.
+
+        # Returns
+        (Object): Elastigroup API response
+        """
+        query_params = self.build_query_params_with_input(
+            {"fromDate": start_date, "toDate": end_date, "limit": limit})
+
+        content = self.send_get(
+            url=self.__base_elastigroup_url + "/" + str(group_id) + "/log",
+            query_params=query_params,
+            entity_name='active events')
+
+        formatted_response = self.convert_json(
+            content, self.camel_to_underscore)
+        return formatted_response["response"]["items"]
+
+    def get_cost_per_account(self, start_date: str, end_date: str):
+        """
+        Retrieve costs up to one year back per specified account over a specified time period.
+
+        # Arguments
+        start_date (String): The start date of the requested time period. The value cannot be earlier than 1 year back.
+        end_date (String): The end date of the requested time period. The maximum of time period length is 90 days.
+
+        # Returns
+        (Object): Cost per Account Response
+        """
+        query_params = self.build_query_params_with_input(
+            {"fromDate": start_date, "toDate": end_date})
+
+        content = self.send_get(
+            url=self.__base_gcp_url + "/costs/",
+            query_params=query_params,
+            entity_name='account cost')
+
+        formatted_response = self.convert_json(
+            content, self.camel_to_underscore)
+        return formatted_response["response"]["items"]
+
+    def get_instance_status(self, instance_id: str):
+        """
+        Get the current instance status. Possible status values: ACTIVE, TERMINATING
+
+        # Arguments
+        instance_id (String): GCP Instance ID
+
+        # Returns
+        (Object) Elastigroup API Response
+        """
+        content = self.send_get(
+            url=self.__base_gcp_url + "/instance/" + instance_id,
+            entity_name='instance status')
+
+        formatted_response = self.convert_json(
+            content, self.camel_to_underscore)
+        return formatted_response["response"]["items"]
+
+    def lock_instance(self, instance_id: str, ttl_in_minutes: str):
+        """
+        Set termination protection for a specific instance.
+
+        # Arguments
+        instance_id (String): GCP Instance ID
+        ttl_in_minutes (String): Specify a TTL (in minutes) for this lock, i.e.: for how long the protection will be valid for.
+
+        # Returns
+        (Object) Spotinst API Response
+        """
+        query_params = dict(ttlInMinutes=ttl_in_minutes)
+
+        response = self.send_post(
```

Comment:
should used send_post_with_params()

Reply:
Done

Reply:
The call gives compilation error at this place.

Reply:
Made back to send_post as post_with_params accepts body as a required aurgument